### PR TITLE
Fix go vet issues

### PR DESCRIPTION
**Pull request by frangarciam**
Fixed all go vet issues and a bunch of golint issues.

### CODE REVIEW

File: agent/metadata_service.go [hunk 1]
```diff
@@ -132,7 +130,7 @@ func (mds *metadataService) getCredentials(w http.ResponseWriter, r *http.Reques
 		Code:            "Success",
 		LastUpdated:     time.Now().UTC().Format(time.RFC3339),
 		Type:            "AWS-HMAC",
-		AccessKeyId:     creds.AccessKeyId,
+		AccessKeyID:     creds.AccessKeyId,
```
Comment (silversupreme) [inline agent/metadata_service.go]: This worries me, as I'm not sure how AWS SDK programs will react to the potential change in the "D" capitalization there. Have you tried this build running on a laptop yet?

Comment (silversupreme) [inline agent/metadata_service.go]: Tested with the Go Playground and it does indeed change the capitalization of the returned JSON object that SDKs will consume. Please try this with a real laptop before you merge.

Comment (frangarciam) [inline agent/metadata_service.go]: Thanks, that's a fair point, and it highlights the reason we were doing these commits in the first place, which is to ensure that the code is easy to work in and testing is comprehensive. In this particular case, we should have been enforcing the format of the json we're outputting (as it's an interface that we need to adhere to), and our tests should include that we're producing the expected json output, which is obviously something we skipped here.

**Comment by zylad** [comment]
LGTM

**Pull request merged**